File: competitive_programming/leetcode/01_python/217_contains_duplicate.py
"""LeetCode 217 - Contains Duplicate."""
def contains_duplicate(nums):
    return len(nums) != len(set(nums))


if __name__ == "__main__":
    assert contains_duplicate([1, 2, 3, 1]) is True
    assert contains_duplicate([1, 2, 3, 4]) is False
    print("217 contains duplicate ok")

File: competitive_programming/leetcode/09_rust/217_contains_duplicate.rs
use std::collections::HashSet;

fn contains_duplicate(nums: &[i32]) -> bool {
    let mut seen = HashSet::new();
    nums.iter().any(|&value| !seen.insert(value))
}

fn main() {
    assert!(contains_duplicate(&[1, 2, 3, 1]));
    assert!(!contains_duplicate(&[1, 2, 3, 4]));
    println!("217 contains duplicate ok");
}
